components/prism-solidity.js

```diff
@@ -0,0 +1,77 @@
+Prism.languages.solidity = Prism.languages.extend('clike', {
+    'keyword': [
+		{pattern: /\b(?:var|import|function|constant|view|pure|payable|storage|memory|if|else|for|while|do|break|continue|returns?|private|public|internal|external|inherited|this|suicide|selfdestruct|emit|new|is|throw|revert|assert|require|\_)\b/},
+        {
+            pattern: /\b(contract|interface|library|using|struct|function|modifier)\s+([A-Za-z_]\w*)(?:\s+is\s+((?:[A-Za-z_][\,\s]*)*))?\b/,
+            inside: {
+                'variable': {
+                    pattern: /\b(contract|interface|library|using|struct|function|modifier)\s+([A-Za-z_]\w*)(?:\s+is\s+((?:[A-Za-z_][\,\s]*)*))?\b/,
+                    lookbehind: true,
+                }
+            }
+		},
+    ],
+	'number': /\b(?:0[xX][\dA-Fa-f]+|0[bB][01]+|0[oO][0-7]+|NaN|Infinity)\b|(?:\b\d+\.?\d*|\B\.\d+)(?:[Ee][+-]?\d+)?/,
+	// Allow for all non-ASCII characters (See http://stackoverflow.com/a/2008444)
+	'function': /[_$a-z\xA0-\uFFFF][$\w\xA0-\uFFFF]*(?=\s*\()/i,
+	'operator': /-[-=]?|\+[+=]?|!=?=?|<<?=?|>>?>?=?|=(?:==?|>)?|&[&=]?|\|[|=]?|\*\*?=?|\/=?|~|\^=?|%=?|\?|\.{3}/,
```

Comment:
Those three lines seem to be exactly the same as the `javascript` component, hence my earlier suggestion: you should probably extend `javascript` instead of `c-like` so that they don't need to be repeated.